compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/Parser.java

```diff
@@ -150,9 +150,9 @@ private void reportSyntaxDiagnostics(BDiagnosticSource diagnosticSource, SyntaxT
             Location syntaxLocation = syntaxDiagnostic.location();
             LineRange lineRange = syntaxLocation.lineRange();
             LinePosition startLine = lineRange.startLine();
-            LinePosition endLine = lineRange.startLine();
-            Location location = new BLangDiagnosticLocation(diagnosticSource.cUnitName, startLine.line() + 1,
-                    endLine.line() + 1, startLine.offset() + 1, endLine.offset() + 1);
+            LinePosition endLine = lineRange.endLine();
+            Location location = new BLangDiagnosticLocation(diagnosticSource.cUnitName, startLine.line(),
+                                                            endLine.line(), startLine.offset(), endLine.offset());
             BLangDiagnostic diag =
                     new BLangDiagnostic(location, syntaxDiagnostic.message(), syntaxDiagnostic.diagnosticInfo());
```

Comment:
With this change, now we don't need to convert a `SyntaxDiagnostic` to `BLangDiagnostic`. Shall we remove that conversion too?

Reply:
Ah ideally yes. Wanted to bring that up actually. That was what I tried at first. But for some reason that broke the sorting of the diagnostics. The syntax diagnostics and semantic diagnostics got interspersed. In our current implementation, we show the syntax errors first and then the semantic errors.